components/metadata/structured-data.tsx

```diff
@@ -0,0 +1,46 @@
+import Script from 'next/script';
+
+interface StructuredDataProps {
+  data: object;
+  id: string;
+}
+
+/**
+ * Component for rendering JSON-LD structured data
+ * Keeps structured data logic separate from page components
+ */
+export function StructuredData({ data, id }: StructuredDataProps) {
+  return (
+    <Script
+      id={id}
+      type="application/ld+json"
+      dangerouslySetInnerHTML={{
+        __html: JSON.stringify(data),
```

Comment:
Potencial issue if there is <script> tag in data
```suggestion
        __html: JSON.stringify(data).replace(/</g, '\\u003c'),
```